external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaDataConsumer.scala

```diff
@@ -587,8 +587,11 @@ private[kafka010] class KafkaDataConsumer(
     case ut: UninterruptibleThread =>
       ut.runUninterruptibly(body)
     case _ =>
-      logWarning("KafkaDataConsumer is not running in UninterruptibleThread. " +
-        "It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894")
+      val warned = _consumer.isDefined
```

Comment:
It looks less intuitive for me. I'm not sure how much this makes us be bothered (the message is quite serious one so I'd be even OK if it's a bit bothering), but if it's really bugging, then I'd add an explicit flag to check it.

Reply:
@HeartSaVioR In micro-batch application, every mini-batch job will generate two logs per task in executor log file. Since micro-batch application is long running, there will be massive warning log.

Reply:
The thing is, the thread should have been UninterruptibleThread; if we found the other case we should file an issue and try to fix.